core/src/validator.rs

```diff
@@ -758,6 +745,21 @@ impl Validator {
             }
         }
 
+        if let Some(start_slot) = should_check_blockstore_for_incorrect_shred_version(
```

Comment:
Although I am confident that we are going to get this logic correct, would it be worthwhile to have a cli argument to perform the scan regardless?

Reply:
> something to pull out in the event we did make a mistake ?

Yeah exactly, if something unforeseen happens in a restart and we are not cleaning up old shreds when we could. However I guess there are only 3 cases, so maybe it's better just to add this as a ledger-tool command instead. That way it could be used in emergencies but also to cleanup ledgers when testing or experimenting with `solana-test-validator`

Reply:
> However I guess there are only 3 cases, so maybe it's better just to add a subcommand to ledger-tool to clean up old shred versions instead

Yeah, can always fall back to ledger-tool; marking this one as resolved for sake of this PR